cpp/include/raft/cluster/kmeans_balanced_types.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/logger.hpp>
+#include <raft/distance/distance_types.hpp>
+#include <raft/random/rng_state.hpp>
+
+namespace raft::cluster::kmeans_balanced {
+
+/**
+ * Simple object to specify hyper-parameters to the balanced k-means algorithm.
+ */
+struct kmeans_balanced_params {
+  /**
+   * Number of training iterations
+   */
+  uint32_t n_iters = 20;
+
+  /**
+   * Metric to use for distance computation.
+   */
+  raft::distance::DistanceType metric = raft::distance::DistanceType::L2Expanded;
```

Comment:
K-means isn't really useful without an L2-based metric. We should probably mention the accepted `DistanceType`s in the docs here.